braindecode/datasets/bids_format.py

```diff
@@ -0,0 +1,714 @@
+        Additional columns to include from descriptions.
+
+    Returns
+    -------
+    pd.DataFrame
+        BIDS-compliant participants DataFrame.
+    """
+    participants_data: dict[str, list[Any]] = {
+        "participant_id": [],
+        "age": [],
+        "sex": [],
+        "hand": [],
+    }
+
+    # Add extra columns
+    extra_data: dict[str, list[Any]] = {col: [] for col in (extra_columns or [])}
+
+    seen_subjects = set()
+
+    for desc in descriptions:
+        if desc is None:
+            continue
+
+        # Get subject ID
+        subject = None
+        for key in ["subject", "sub", "subject_id"]:
+            if key in desc.index and pd.notna(desc[key]):
+                subject = str(desc[key])
+                break
+
+        if subject is None:
+            continue
+
+        # Skip duplicates
+        if subject in seen_subjects:
+            continue
+        seen_subjects.add(subject)
+
+        # Format as BIDS participant_id
+        participant_id = f"sub-{subject}" if not subject.startswith("sub-") else subject
+
+        # Get other info
+        age = desc.get("age", "n/a")
+        sex = desc.get("sex", desc.get("gender", "n/a"))
+        hand = desc.get("hand", desc.get("handedness", "n/a"))
+
+        participants_data["participant_id"].append(participant_id)
+        participants_data["age"].append(age if pd.notna(age) else "n/a")
+        participants_data["sex"].append(sex if pd.notna(sex) else "n/a")
+        participants_data["hand"].append(hand if pd.notna(hand) else "n/a")
+
+        # Add extra columns
+        for col in extra_columns or []:
+            extra_data[col].append(desc.get(col, "n/a"))
+
+    # Combine all data
+    participants_data.update(extra_data)
+
+    return pd.DataFrame(participants_data)
+
+
+def create_channels_tsv(
+    info: "mne.Info",
+    bad_channels: Optional[list[str]] = None,
+) -> pd.DataFrame:
+    """
+    Create a BIDS-compliant channels.tsv from MNE Info.
+
+    Parameters
+    ----------
+    info : mne.Info
+        MNE Info object containing channel information.
+    bad_channels : list of str | None
+        List of bad channel names.
+
+    Returns
+    -------
+    pd.DataFrame
+        BIDS-compliant channels DataFrame.
+    """
+    import mne
+
+    bad_channels = bad_channels or info.get("bads", [])
+
+    channels_data: dict[str, list[Any]] = {
+        "name": [],
+        "type": [],
+        "units": [],
+        "sampling_frequency": [],
+        "low_cutoff": [],
+        "high_cutoff": [],
+        "status": [],
+        "status_description": [],
+    }
+
+    # MNE channel type to BIDS type mapping
+    type_mapping = {
+        "eeg": "EEG",
+        "ecg": "ECG",
+        "eog": "EOG",
+        "emg": "EMG",
+        "meg": "MEG",
+        "ref_meg": "MEGREF",
+        "grad": "MEGGRADAXIAL",
+        "mag": "MEGMAG",
+        "stim": "TRIG",
+        "misc": "MISC",
+        "bio": "BIO",
+    }
```

Comment:
From the mne-bids function [`_channels_tsv()`](https://github.com/mne-tools/mne-bids/blob/main/mne_bids/write.py#L161C5-L183C6)
It seems these are the correct channel names:
```suggestion
    type_mapping = defaultdict(lambda: "Other type of channel")
    type_mapping.update(
        meggradaxial="Axial Gradiometer",
        megrefgradaxial="Axial Gradiometer Reference",
        meggradplanar="Planar Gradiometer",
        megmag="Magnetometer",
        megrefmag="Magnetometer Reference",
        stim="Trigger",
        eeg="ElectroEncephaloGram",
        ecog="Electrocorticography",
        seeg="StereoEEG",
        ecg="ElectroCardioGram",
        eog="ElectroOculoGram",
        emg="ElectroMyoGram",
        misc="Miscellaneous",
        bio="Biological",
        ias="Internal Active Shielding",
        dbs="Deep Brain Stimulation",
        fnirs_cw_amplitude="Near Infrared Spectroscopy (continuous wave)",
        resp="Respiration",
        gsr="Galvanic skin response (electrodermal activity, EDA)",
        temperature="Temperature",
    )
```
It would have been great to use the functions already implemented in mne-bids but they only work on raw objects, not epochs